dbt/include/athena/macros/materializations/incremental.sql

```diff
@@ -48,9 +48,9 @@
     {%- set single_partition_expression = single_partition | join(' and ') -%}
     {%- do partitions.append('(' + single_partition_expression + ')') -%}
   {%- endfor -%}
-  {%- set partition_expression = partitions | join(' or ') -%}
-  {%- do adapter.clean_up_partitions(target_relation.schema, target_relation.table, partition_expression) -%}
-
+  {%- for i in range(partitions | length) %}
+    {%- do adapter.clean_up_partitions(target_relation.schema, target_relation.table, partitions[i]) -%}
```

Comment:
Hi @ojasviharsola Thanks for the PR and sorry for the late response. Changing this way results in a call to Glue API and S3 API for every partitions to be deleted, which is very inefficient. 

For your use case, how many partitions and partition keys were you trying to clean up to get to the limit?  I'm wondering how often you get to that limit. Is there a workaround, like cleaning up target table in advance?